src/main/java/com/Duke/TaskManager/TaskList.java

```diff
@@ -0,0 +1,57 @@
+package com.Duke.TaskManager;
+
+import com.Duke.Tasks.Task;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskList {
+    //This class manages the list of Tasks that the user has added
+    public final List<Task> ls;
+
+    public TaskList(List<Task> ls) {
+        this.ls = ls;
+    }
+
+    public int length(){
+        return ls.size();
+    }
+
+    public void setDone(int task) throws DukeException {
+        if(task>ls.size()){
+            throw new DukeException("You don't have that many tasks");
+        }else {
+            ls.set(task, ls.get(task).done());
+        }
+    }
+
+    public TaskList findTask(String hint){
+        TaskList containsHint = new TaskList(new ArrayList<Task>());
+        for(Task task: ls){
+            if(task.getTask().contains(hint)){
+                containsHint.add(task);
+            }
+        }
+        return containsHint;
+    }
+
+    public List<Task> getListOfTasks(){
+        return this.ls;
+    }
+
+    public void delete(int task) throws DukeException{
+        if(task>ls.size()){
```

Comment:
You might want to add some spacing here. I noticed this in other places as well perhaps you can look through your code and change them accordingly.